### DISCUSSION TITLE
does Date.now() value is based on machine it runs on, or it works as nodejs it gives the value from epoch??

### DISCUSSION THREAD

**Discussion by mhmadalaa** [Automattic/mongoose#14440] [Q&A] [answered]
i create a new instance in database, there are two fields: the first is filled by `default: Date.now() ` in mongoose schema. and the scond is filled by `Date.now()` the nodejs function 
the two answers is different??
so in mongoose it depeond on the machine right??
and how can i set the default not to do that behavior

**Comment by vkarpov15** [comment, marked as answer]
`default: Date.now()` vs `default: () => Date.now()` or `default: Date.now`?

That's because `default: Date.now()` sets the default to the current value of `Date.now()`. So equivalent to `Default: 1710793129346`. Basically you're setting the default value to your server's start time.

With `default: () => Date.now()`, you're setting `default` to a function that Mongoose executes to get the default each time.
